app/src/main/java/com/sopt/dive/signup/SignUpActivity.kt

```diff
@@ -0,0 +1,281 @@
+                        unfocusedContainerColor = Color.Transparent,
+                        unfocusedPlaceholderColor = Color.LightGray
+                    ),
+                    singleLine = true,
+                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Next),
+                    keyboardActions = KeyboardActions(onNext = {
+                        focusManager.moveFocus(FocusDirection.Down)
+                    }),
+                    placeholder = {
+                        Text(text = stringResource(R.string.signup_nickname_hint))
+                    },
+                    modifier = Modifier.fillMaxWidth()
+                )
+
+                Spacer(Modifier.height(24.dp))
+
+                /** Drink **/
+                Text(
+                    text = stringResource(R.string.signup_drink_title),
+                    fontSize = 28.sp,
+                    fontWeight = FontWeight.Normal,
+                    color = Color.Black
+                )
+
+                TextField(
+                    value = drinkText,
+                    onValueChange = { drinkText = it },
+                    colors = TextFieldDefaults.colors(
+                        focusedContainerColor = Color.Transparent,
+                        unfocusedContainerColor = Color.Transparent,
+                        unfocusedPlaceholderColor = Color.LightGray
+                    ),
+                    singleLine = true,
+                    keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done),
+                    keyboardActions = KeyboardActions(onDone = {
+                        focusManager.clearFocus()
+                    }),
+                    placeholder = {
+                        Text(text = stringResource(R.string.signup_drink_hint))
+                    },
+                    modifier = Modifier.fillMaxWidth()
+                )
+
+                Spacer(Modifier.height(32.dp))
+
+                Spacer(Modifier.weight(1f))
+            }
+        }
+    }
+    
+    private fun checkValidation(id: String, pw: String, nickName: String): Boolean {
+        if(id.isEmpty() || pw.isEmpty() || nickName.isEmpty()) {
+            Toast.makeText(this, this.resources.getString(R.string.toast_signup_unavailable), Toast.LENGTH_SHORT).show()
+            return false
+        } else if(id.length !in 6..10) {
+            Toast.makeText(this, this.resources.getString(R.string.toast_signup_id_unavailable), Toast.LENGTH_SHORT).show()
+            return false
+        } else if(pw.length !in 8..12) {
+            Toast.makeText(this, this.resources.getString(R.string.toast_signup_pw_unavailable), Toast.LENGTH_SHORT).show()
+            return false
+        } else if(nickName.isEmpty()) {
+            Toast.makeText(this, this.resources.getString(R.string.toast_signup_nickname_unavailable), Toast.LENGTH_SHORT).show()
+            return false
+        } else {
+            return true
+        }
+    }
```

Comment:
```suggestion
private fun checkValidation(id: String, pw: String, nickName: String): Boolean {
    val messageRes = when {
        id.isEmpty() || pw.isEmpty() || nickName.isEmpty() ->
            R.string.toast_signup_unavailable
        id.length !in 6..10 ->
            R.string.toast_signup_id_unavailable
        pw.length !in 8..12 ->
            R.string.toast_signup_pw_unavailable
        nickName.isEmpty() ->
            R.string.toast_signup_nickname_unavailable
        else -> null
    }

    return if (messageRes != null) {
        Toast.makeText(this, getString(messageRes), Toast.LENGTH_SHORT).show()
        false
    } else { 
        true
    }
}
```
다음과 같이 Toast 만 되는 구조라면 이렇게 작성하면 중복을 줄일 수 있을 것 같습니다.